python/jupytergis_lab/jupytergis_lab/notebook/gis_document.py

```diff
@@ -59,17 +57,16 @@ def __init__(
         pitch: Optional[float] = None,
         projection: Optional[str] = None,
     ):
-        if isinstance(path, str):
-            path = Path(path)
-
-        self.path = path
-
-        comm_metadata = GISDocument._path_to_comm(str(self.path) if self.path else None)
+        if isinstance(path, Path):
+            path = str(path)
 
         ydoc = Doc()
 
         super().__init__(
-            comm_metadata=dict(ymodel_name="@jupytergis:widget", **comm_metadata),
+            comm_metadata={
+                "ymodel_name": "@jupytergis:widget",
+                **self._path_to_comm(path),
```

Comment:
Instance methods can call class methods from `self`, which is less brittle than calling it from `GISDocument`.

Reply:
Changing from `dict()` to `{}` was just personal preference. No strong feeling.